haystack/utils/preprocessing.py

```diff
@@ -6,7 +7,7 @@
 
 from haystack.nodes.file_converter import BaseConverter, DocxToTextConverter, PDFToTextConverter, TextConverter
 from haystack.schema import Document
-
+from haystack.nodes.preprocessor import PreProcessor
```

Comment:
I'd argue the changes to `convert_files_to_docs`  should be removed from this PR. Let's just delete `tika_convert_files_to_docs`: is a legacy function, fully unused (I was even unaware of its existence :smile:)

The changes themselves are ok: this is a problem of scope. `convert_files_to_docs` is a simple function used by some tutorials to hide the indexing process from newbies. This is why only a few file types were hard-coded inside. It's not supposed to actually do proper indexing with Tika and preprocessors.

If you're at the point where you need proper indexing, please rather consider using an indexing pipeline.

https://haystack.deepset.ai/components/pipelines

Let me know if anything is unclear.

Reply:
I don't like to tell contributors to delete code so I get your point... I wouldn't like to be told that either.

But I do believe that this code will be removed anyway in the next weeks. We have the removal of `convert_files_to_docs` on our roadmap, to give you an idea of how close is this whole part of the framework to be disposed of (not done yet only because it's heavily used in the tutorials, and they will require a serious rework). I don't think it's worth adding these changes for such a short time, it might even be dangerous. It gives users the wrong signal and might turn someone into thinking these functions are maintained, rely on them, only to be disappointed when we remove them. Really sorry for this.